src/main/java/org/lanternpowered/server/entity/LanternLiving.java

```diff
@@ -104,18 +127,89 @@ public void pulse() {
                 if (potionEffect.getType() == PotionEffectTypes.INVISIBILITY) {
                     offer(Keys.INVISIBLE, duration > 0);
                 }
+                if (potionEffect.getType() == PotionEffectTypes.HUNGER && supports(Keys.EXHAUSTION)) {
+                    offer(Keys.EXHAUSTION, Math.min(get(Keys.EXHAUSTION).get() + (0.005 * (potionEffect.getAmplifier() + 1)),
+                            getValue(Keys.EXHAUSTION).get().getMaxValue()));
+                }
             }
             offer(Keys.POTION_EFFECTS, newPotionEffects.build());
         }
     }
 
-    @Override
-    public <T extends Projectile> Optional<T> launchProjectile(Class<T> projectileClass) {
-        return Optional.empty();
+    private void pulseFood() {
+        if (!get(FoodData.class).isPresent()) {
+            return;
+        }
+        final Difficulty difficulty = getWorld().getDifficulty();
+
+        if (get(Keys.EXHAUSTION).get() > 4.0) {
+            offer(Keys.EXHAUSTION, get(Keys.EXHAUSTION).get() - 4.0);
+
+            final MutableBoundedValue<Double> saturation = getValue(Keys.SATURATION).get();
+
+            if (saturation.get() > saturation.getMaxValue()) {
+                offer(Keys.SATURATION, Math.max(saturation.get() - 1.0, saturation.getMinValue()));
+            } else if (!difficulty.equals(Difficulties.PEACEFUL)) {
+                offer(Keys.FOOD_LEVEL, Math.max(get(Keys.FOOD_LEVEL).get() - 1, getValue(Keys.FOOD_LEVEL).get().getMinValue()));
+            }
+        }
+
+        final boolean naturalRegeneration = getWorld().getGameRule(DefaultGameRules.NATURAL_REGENERATION).orElse("true").equals("true");
```

Comment:
Use the `getRule` method, this will automatically handle parsing. The rule can be found in `RuleTypes`.